server/api/crud/runtimes/nuclio/function.py

```diff
@@ -453,13 +458,18 @@ def _set_build_params(function, nuclio_spec, builder_env, project, auth_info=Non
         )
 
 
-def _set_function_scheduling_params(function, nuclio_spec):
+def _set_function_scheduling_params(function, nuclio_spec, db_session):
     # don't send node selections if nuclio is not compatible
     if mlrun.runtimes.nuclio.function.validate_nuclio_version_compatibility(
         "1.5.20", "1.6.10"
     ):
         if function.spec.node_selector:
-            nuclio_spec.set_config("spec.nodeSelector", function.spec.node_selector)
+            nuclio_spec.set_config(
+                "spec.nodeSelector",
+                _enrich_function_node_selector_with_project(
+                    db_session, function.metadata.project, function.spec.node_selector
+                ),
+            )
```

Comment:
what if `function.spec.node_selector` is empty but `project.spec.default_function_node_selector` is not empty?
resolve the node selector first, then check:
```suggestion
        if node_selector := _enrich_function_node_selector_with_project(
                    db_session, function.metadata.project, function.spec.node_selector
                ):
                nuclio_spec.set_config(
                "spec.nodeSelector",
                node_selector,
            )
```